File: Ahorn/entities/SulfurVent.jl
module SulfurVent

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/SulfurVent" SulfurVent(x::Integer, y::Integer)

const placements = Ahorn.PlacementDict(
    "normal" => Ahorn.EntityPlacement(SulfurVent)
)

function Ahorn.selection(entity::SulfurVent)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::SulfurVent, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end

File: framework/extracted/Loenn/entities/sulfurVent.lua
local utils = require("utils")
local drawableSprite = require("structs.drawable_sprite")

local sulfurVent = {}

sulfurVent.name = "MaggyHelper/SulfurVent"
sulfurVent.depth = -50
sulfurVent.placements = {
    {
        name = "normal",
        data = {}
    }
}

function sulfurVent.sprite(room, entity)
    local sprite = drawableSprite.fromTexture("objects/sulfur_vent/default", entity)
    sprite:setPosition(entity.x, entity.y)
    sprite:setJustification(0.5, 1.0)
    sprite.color = {0.8, 0.8, 0.4, 1.0}
    return sprite
end

function sulfurVent.selection(room, entity)
    return utils.rectangle(entity.x - 8, entity.y - 12, 16, 12)
end

return sulfurVent
